### ISSUE TITLE
Running out of Memory

### ISSUE THREAD

**Issue by antonydanby** [microsoft/vscode-mssql#777]
- MSSQL Extension Version: 0.3.0
- VSCode Version: 1.10.2
- OS Version: Windows 10 64 bit 

My system ground to a halt this morning. When I looked at Task Manager there was a dotnet process at 100% and consuming over 2Gb ( Yes 2GB; I didn't make a mistake  ). When I switched on command line, it was your mssql extension going absolutely nuts. I had no choice but to kill the task before it killed my system.

Must admit I am always shocked when coders write code for other coders that misbehaves !!

Steps to Reproduce:

I didn't do anything, except open VSCODE and start editing. It spun off it's own axis about 30 minutes into my editing session.

The only things that might be a factor are :
1. I had several SQL files open in VSCODE in several instances of VSCODE; not just 1
2. I hibernated my laptop and opened it back up this morning

**Comment by benrr101** [comment]
Hi @antonydanby thanks for your feedback. Agreed, it is frustrating when software by developers for developers misbehaves -- but unfortunately, we're not infallible.

To help me get to the bottom of this issue, can you provide us some more details about the scenario? Had you connected any of your editors to a database before you let it sit? Had you executed any queries in the editors? Have you been able to reliably reproduce this issue?

I'm also unclear what you mean by "it spun off it's own axis." Can you clarify that statement a bit?

Again thanks for raising this issue, and we'll get to the bottom of it.

**Comment by antonydanby** [comment]
Sorry about that I'll try to avoid using flowery prose. when I said "spun of it's own axis" I meant that it was consuming so much resources and CPU my machine had ground to an almost standstill; invoking Task Manager at this point was taking 10 to 20 seconds just for it to appear.

The process was:

C:\Users\Danbya\.vscode\extensions\ms-mssql.mssql-0.3.0\sqltoolsservice\0.3.0.5\Windows\Microsoft.SqlTools.ServiceLayer.exe

The editors, I suspect, although I can't be sure, were still connected over a hibernate from the night before. I hadn't executed any queries in the morning prior to the point when it started to consume  high amounts of resources / RAM. 

I havn't be able to artificially induce this condition again and it hasn't re-appeared today and I hibernated last night with editors still open to see if it would occur again and it didn't ( so far )

Hope this helps

Thanks
Tony Danby

**Comment by benrr101** [comment]
Thanks for the clarification! We've had a report of an issue like this before. I'm going to spin up a VM to try and reproduce the issue, but since it seems so intermittent, it might be hard to reproduce. Nevertheless, we want to figure something this serious out.

**Comment by victorschinaider** [comment]
I'm having the same problem in Ubuntu. After use the mssql extension, the pc totally freezes. Memory jump from 2 GB of consumption, to 7,5 GB.

**Comment by kevcunnane** [comment]
@benrr101 we've had 2 reports today so might be good to reengage on this.

**Comment by IAmHopp** [comment]
I just ran into this as well. I'm going to share some info hoping it'll be helpful:

First of all, I just installed this extension about an hour ago to edit one file. So, just to highlight some key differences to @antonydanby's situation: This extension definitely doesn't need hours [as, for example, over a hibernate] to behave this way and it also doesn't need multiple SQL files.

The file I was editing is about 12000 lines long. It's a big insert statement. Here's what I did:

1. Opened the file.
2. Ran the query. [Creating a connection profile in the process, so at this point I have exactly one connection to exactly one database.]
3. Got an error. [A pure SQL error, just to highlight nothing actually ran in the database at this point.]
4. Fixed the error.
5. Ran the query again.
6. Got an error about the INSERT with more than 1000 lines scenario.
7. Used multiple cursors to break the file into multiple one-line inserts.

When I finished editing the file, the PC felt sluggish, so I took a look at task manager and MicrosoftSqlToolsServiceLayer.exe was sitting at 5GB. I closed the file, but left Code open to see if it'd cool down. About half an hour has passed so far and it's still sitting at 4.8GB.

Again, nothing actually ran in the database. I did run the query twice, if that makes any difference, but got an error two times. I did not try to execute after inserting the file.

Summing up: Just one file, just one connection, half an hour for the issue to manifest, no query ran successfully, file is not open in the editor.

I know issues like this are very difficult to pin down, but as an user that just started using this today with a PC now sitting at 99% memory usage because of one process, I have to say it seems pretty serious.

Anyway, appreciate you guys for all the help, sorry for the long post. I wanted to get as many details in as possible.

**Comment by CrossTyreck** [comment]
Since this original post is still opened I decided to repost back here. 

I have had the same issue. I am on SQL Server v.13.0.4411.0.

I believe my issue is centered around a certain file though. Here are my observations:

If I load a specific stored procedure with a lot of SQL (1800+ lines) memory just sits at 19.3 MB.

When I connect to a db (no execution just connecting) the memory usage starts climbing and bounces back and forth for 20 min till it finally taps out my system resources and I have to disable the extension and reload the project.

If I remove all the lines from the file after connecting to the db the memory still continues to climb.

I also have multiple cursors peppered through out the file, some nested. 

I see the same behavior as @IAmHopp has identified.

**Comment by kburtram** [comment]
@MattIrv could you please take a look at this one again and see if we can get a local repro or otherwise diagnosis the problem.

**Comment by thanhhungtn1** [comment]
Just open and check the process on Mac OS Mojave - 15 GB RAM is using by this process. 
That's a lot :-/

**Comment by aleksandro83** [comment]
Any update about this?
The process which causes the memory climb is MicrosoftSqlToolsServiceLayer.exe.
I've notice this problem occurs in vscode with mssql extension, sql operations studio and azure data studio, all related with MicrosoftSqlToolsServiceLayer.

Could somebody please take a look at this?

**Comment by abist** [comment]
Please feel free to reopen the issue if you still see the problem persist in the new version of the extension.
